src/dotnet/CarbonAware.DataSources.WattTime/WattTimeDataSource.cs

```diff
@@ -0,0 +1,74 @@
+﻿using CarbonAware.Interfaces;
+using CarbonAware.Model;
+using CarbonAware.Tools.WattTimeClient;
+using Microsoft.Extensions.Logging;
+using System.Diagnostics;
+using System.Globalization;
+
+namespace CarbonAware.DataSources.WattTime;
+
+/// <summary>
+/// Reprsents a wattime data source.
+/// </summary>
+public class WattTimeDataSource : ICarbonIntensityDataSource
+{
+    private ILogger<WattTimeDataSource> Logger { get; }
+
+    private IWattTimeClient WattTimeClient { get; }
+
+    private ActivitySource ActivitySource { get; }
+
+    private IRegionConverter RegionConverter { get; }
+
+    /// <summary>
+    /// Creates a new instance of the <see cref="WattTimeDataSource"/> class.
+    /// </summary>
+    /// <param name="logger">The logger for the datasource</param>
+    /// <param name="client">The WattTime Client</param>
+    /// <param name="activitySource">The activity source for telemetry.</param>
+    /// <param name="regionConverter">The region converter to be used to convert Azure regions to BA's.</param>
+    public WattTimeDataSource(ILogger<WattTimeDataSource> logger, IWattTimeClient client, ActivitySource activitySource, IRegionConverter regionConverter)
+    {
+        this.Logger = logger;
+        this.WattTimeClient = client;
+        this.ActivitySource = activitySource;
+        this.RegionConverter = regionConverter;
+    }
+
+    /// <inheritdoc />
+    public async Task<IEnumerable<EmissionsData>> GetCarbonIntensityAsync(string region, DateTimeOffset startPeriod, DateTimeOffset endPeriod)
+    {
+        this.Logger.LogInformation("Getting carbon intensity for region {region} for period {startPeriod} to {endPeriod}.", region, startPeriod, endPeriod);
+
+        using (var activity = ActivitySource.StartActivity())
```

Comment:
I'm sure an over the shoulder review would be helpful to me here but can someone remind me what the activity source does?

Reply:
It's for telemetry.  Specifically traces.